lib/r10k/git/errors.rb

```diff
@@ -10,24 +10,24 @@ class UnresolvableRefError < GitError
       attr_reader :ref
       attr_reader :git_dir
 
-      def initialize(*args)
+      def initialize(mesg, options = {})
+        if mesg.is_a? Hash
```

Comment:
It's a quick of having variable arity methods in ruby, where callers can provide a message, additional options, or both. If both are passed then it's easy, but if just an options hash is passed and not a message, then all arguments will be shifted down - mesg will be the hash, and options will be nil.

That being said, I think this is pretty messy after looking at it again and should be revised.